pkg/KubeArmorAnnotation/handlers/pod_mutation.go

```diff
@@ -113,6 +114,27 @@ func (a *PodAnnotator) InjectDecoder(d *admission.Decoder) error {
 // == Add AppArmor annotations == //
 func appArmorAnnotator(pod *corev1.Pod) {
 	podAnnotations := map[string]string{}
+	var podOwnerName string
+
+	// podOwnerName is the pod name for static pods and parent object's name
+	// in other cases
+	for _, ownerRef := range pod.ObjectMeta.OwnerReferences {
```

Comment:
Can we use the newly created `GetControllingPodOwner()` to replace this loop?

Reply:
I couldn't find any example where we were using something from kubearmor's packages in any of the controllers so doing it here kind of felt like an anti-pattern.
Also, the annotations controller and kubearmor use different k8s API versions so we'll have to update the API version used in the controller if we import something or we'll face bugs while building. I thought it would further complicate this PR :sweat_smile: